.ci/scripts/test_global_patch.py

```diff
@@ -18,52 +18,84 @@
 import subprocess
 import sys
 
-# test patching from command line
-err_code = subprocess.call(
-    [sys.executable, "-m", "sklearnex.glob", "patch_sklearn", "-a", "svc"]
-)
-assert not err_code
-from sklearn.svm import SVC, SVR
-
-assert SVC.__module__.startswith("daal4py") or SVC.__module__.startswith("sklearnex")
-assert not SVR.__module__.startswith("daal4py") and not SVR.__module__.startswith(
-    "sklearnex"
-)
-
-
-from sklearnex import patch_sklearn, unpatch_sklearn
-
-# test unpatching from command line
-err_code = subprocess.call([sys.executable, "-m", "sklearnex.glob", "unpatch_sklearn"])
-assert not err_code
-unpatch_sklearn()
-from sklearn.svm import SVC, SVR
-
-assert not SVC.__module__.startswith("daal4py") and not SVC.__module__.startswith(
-    "sklearnex"
-)
-assert not SVR.__module__.startswith("daal4py") and not SVR.__module__.startswith(
-    "sklearnex"
-)
-
-
-# test patching from function
-patch_sklearn(name=["svc"], global_patch=True)
-from sklearn.svm import SVC, SVR
-
-assert SVC.__module__.startswith("daal4py") or SVC.__module__.startswith("sklearnex")
-assert not SVR.__module__.startswith("daal4py") and not SVR.__module__.startswith(
-    "sklearnex"
-)
-
-
-# test unpatching from function
-unpatch_sklearn(global_unpatch=True)
-from sklearn.svm import SVC, SVR
-
-assert not SVC.__module__.startswith("daal4py") and not SVC.__module__.startswith(
-    "sklearnex"
-)
-assert not SVR.__module__.startswith("daal4py") and not SVR.__module__.startswith(
-    "sklearnex"
-)
+import pytest
+
+
+@pytest.fixture
+def patch_from_command_line(request):
+    err_code = subprocess.call(
+        [sys.executable, "-m", "sklearnex.glob", "patch_sklearn", "-a", "svc"]
+    )
+    assert not err_code
+
+    def unpatch_from_cmd():
+        err_code = subprocess.call(
+            [sys.executable, "-m", "sklearnex.glob", "unpatch_sklearn"]
+        )
+        assert not err_code
+
+    request.addfinalizer(unpatch_from_cmd)
+    return
+
+
+def test_patching_from_command_line(patch_from_command_line):
+    from sklearn.svm import SVC, SVR
+
+    assert SVC.__module__.startswith("daal4py") or SVC.__module__.startswith("sklearnex")
+    assert not SVR.__module__.startswith("daal4py") and not SVR.__module__.startswith(
+        "sklearnex"
+    )
+
+
+def test_unpatching_from_command_line(patch_from_command_line):
+    err_code = subprocess.call(
+        [sys.executable, "-m", "sklearnex.glob", "unpatch_sklearn"]
+    )
+    assert not err_code
+    from sklearnex import unpatch_sklearn
+
+    unpatch_sklearn()

```

Comment:
https://github.com/intel/scikit-learn-intelex/blob/main/sklearnex/tests/test_monkeypatch.py#L65  Just be careful with these functions since they impact global state and aren't local to the test (anything with a patch_sklearn call should be with a 'try-finally' with an unpatch_sklearn)

Reply:
Yes, that's the idea behind the fixtures with finalizers.